algorithms/ar-bsrh/csharp/BinarySearch.cs

```diff
@@ -0,0 +1,45 @@
+// C# implementation of Binary Search 
+using System; 
+
+class BinarySearch { 
```

Comment:
Codacy found an issue: [Add a 'protected' constructor or the 'static' keyword to the class declaration.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6330839)